x/btcstaking/keeper/msg_server.go

```diff
@@ -165,6 +165,11 @@ func (ms msgServer) BtcStakeExpand(goCtx context.Context, req *types.MsgBtcStake
 		return nil, status.Errorf(codes.InvalidArgument, "the previous BTC staking transaction FP: %+v is not the same as FP of the stake expansion %+v", prevBtcDel.FpBtcPkList, req.FpBtcPkList)
 	}
 
+	// check that the previous delegation and the new expansion has the same staker btc pk
+	if err := validateStakerBtcPks(parsedMsg, prevBtcDel); err != nil {
```

Comment:
@GAtom22 shouldn't we validate that right now at main? the btc_pks of previous stk and extended stake should match